doc/technotes/errorHandling.rst

```diff
@@ -0,0 +1,190 @@
+.. _readme-errorHandling:
+
+==============
+Error Handling
+==============
+
+This README describes the error handling feature as currently implemented.
+
+
+Errors
+------
+A base class ``Error`` is defined in the Chapel standard modules.
+
+::
+
+  class Error {
+    var msg: string;
+
+    proc Error(_msg: string) {
+      msg = _msg;
+    }
+  }
+
+Users may use the provided ``Error`` directly, or create their own error
+hierarchy with ``Error`` at its root.
+
+::
+
+  class MyError: Error {}
+
+  class MyIntError: MyError {
+    var i: int;
+  }
+
+Chapel does not provide a standard set of ``Error`` classes.
+
+
+Throwing Errors
+---------------
+``throw`` statements throw errors from a function to its callee.
+
+::
+
+  throw new Error("with a message");
+
+  var e: Error;
+  // ... set e ...
+  throw e;
+
+If a function may throw an error, it must be declared with ``throws``.
+
+::
+
+  proc mayThrowErrors() throws { ... }
+  proc mayThrowErrorsAlso(): A throws where ... { ... }
+  proc mayNotThrowErrors() { ... }
+
+
+Handling Errors
+---------------
+
+Default Policy
+++++++++++++++
+
+When a throwing function is called, the default error handling policy of the
+calling function is as follows:
```

Comment:
Is there some better way to describe this than "calling function" and "called function"? These terms are visually very similar. I think what would help the most would be to illustrate what you mean with a code sample like this:

``` chapel
  proc inFunction() {
     .... call() ...
  }
```

What does `call` do with respect to error handling? 


But... maybe there are other terms that would work nicely... I don't disagree with anything you're writing, just wonder if it will be hard to follow right here.